File: typescript/day1/prb.ts
const content = await Deno.readTextFile("../../inputs/d1");

const lines = content.split(["\n"]).map((x: string) => x.trim());
const values = lines.map((x: string) =>
  x.split(" ").map((x: string) => x.trim())
)
  .flat()
  .filter((x: string) => x != "");

const left: number[] = [];
const right: number[] = [];
values.forEach((x: string, index: number) => {
  if (index % 2 == 0) {
    right.push(parseInt(x));
  } else {
    left.push(parseInt(x));
  }
});

left.sort();
right.sort();

function part1(left: number[], right: number[]): number {
  let count = 0;
  left.forEach((x: number, index: number) => {
    count += Math.abs(x - right[index]);
  });
  return count;
}
function part2(left: number[], right: number[]): number {
  let acc = 0;
  left.forEach((x: number) => {
    acc += x * right.filter((v: number) => x == v).length;
  });
  return acc;
}

console.log(`Part1: ${part1(left, right)}`);
console.log(`Part2: ${part2(left, right)}`);

File: python/day1/prb.py
left = []
right = []
with open("./input", "r") as f:
    lines = f.readlines()
    for line in lines:
        v1, v2 = line.strip().split("   ")
        left.append(int(v1))
        right.append(int(v2))


def part1(v1, v2):
    acc = 0
    acc = sum((abs(a - b) for a, b in zip(v1, v2)))
    return acc


def part2(left, right):
    acc = sum((a * right.count(a) for a in left))
    return acc


p1 = part1(sorted(left), sorted(right))
p2 = part2(left, right)

print(f"part1: {p1}")
print(f"part2: {p2}")
